components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -332,16 +389,23 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
     f.get_header_ptr()->get_tracking().add_customer(weak_from_this());
   }
 
-  virtual void set_required_field_impl (const Field<const Real>& f) = 0;
-  virtual void set_computed_field_impl (const Field<      Real>& f) = 0;
+  virtual void set_required_field_impl (const Field<const Real>& f) {};
+  virtual void set_computed_field_impl (const Field<      Real>& f) {};
 
-private:
+  virtual void check_required_fields_impl () const {}
```

Comment:
Here too: the base class already runs all checks, so in general there's no need to override these. But derived class _may_ want to do more stuff. E.g., repair the field if the checks don't pass, or, for a AdaptiveAtmProcess, we may think of "re-doing" the timestep, subcycling with dt/2.